src/3d/qgs3daxis.cpp

```diff
@@ -348,20 +364,34 @@ QVector3D Qgs3DAxis::from3dTo2dLabelPosition( const QVector3D &sourcePos,
   destPos.setY( destPos.y() - viewTranslation.y() );
   destPos.setZ( 0.0f );
 
-#ifdef DEBUG
-  qDebug() << "from3dTo2dLabelPosition: sourcePos" << sourcePos << " with" << viewTranslation << "corrected destPos" << destPos;
-#endif
+  if ( QgsLogger::isTraceEnabled() )
+  {
+    std::ostringstream os;
+    os << "Qgs3DAxis::from3DTo2DLabelPosition: sourcePos: " << sourcePos.toPoint()
+       << " with translation: " << viewTranslation.toPoint()
+       << " corrected to pos: " << destPos.toPoint();
+    QgsTraceMsg( os.str().c_str() );
+  }
   return destPos;
 }
 
 void Qgs3DAxis::setEnableCube( bool show )
 {
   mCubeRoot->setEnabled( show );
+  if ( show )
+    mCubeRoot->setParent( mAxisSceneEntity );
```

Comment:
I prefer with brackets, even for one line statement :stuck_out_tongue: 

```
if (x)
{
...
}
```

Reply:
fixed!